js/script.js

```diff
@@ -99,13 +169,65 @@ document.querySelector('#book-form').addEventListener('submit', (e) => {
   const title = document.querySelector('#title').value;
   const author = document.querySelector('#author').value;
   const isbn = document.querySelector('#isbn').value;
+  const email = document.querySelector('#authorEmail').value;
+  const phone = document.querySelector('#author-phone').value;
 
   // Validate
-  if(title === '' || author === '' || isbn === '') {
-    UI.showAlert('Please fill in all fields', 'danger');
+  if (title === '') {
+    UI.show_error_text('.titleError');
+  }
+  if (author === '') {
+    UI.show_error_text('.authorError');
+  }
+  if (isbn === '') {
+    UI.show_error_text('.isbnError');
+  }
+  if (email === '') {
+    UI.show_error_text('.emailError');
+  }
+  if (phone === '') {
+    UI.show_error_text('.phoneError');
+  }
+
+  if(title === '' || author === '' || isbn === '' || email === '') {
+    return;
```

Comment:
This can be extracted into a separate validation function as well. This will also remove the somewhat redundant checks on line 192. It would change to:

```
if ( !isValid(book) ) {
  return;
}else{
```

Breaking down large functions into smaller ones also makes them more testable.

Reply:
Also there is no validation for uniqueness of ISBN. Please add.